pwiz_tools/Skyline/TestPerf/PerfUniquePeptidesTest.cs

```diff
@@ -78,43 +78,43 @@ void LogToConsole(string message, [CallerLineNumber] int lineNumber = 0)
             }
         }
 
-        [TestMethod, NoParallelTesting(TestExclusionReason.RESOURCE_INTENSIVE)]
+        [TestMethod, NoParallelTesting(TestExclusionReason.RESOURCE_INTENSIVE), Timeout(TEST_TIMEOUT_MSEC)]
         public void UniquePeptides0PerfTest()
         {
             // Scenarios to test:
             // 0) Background proteome nicely digested and metatdata resolved
             scenario(PeptideFilter.PeptideUniquenessConstraint.none, "human_and_yeast.protdb");
         }
 
-        [TestMethod, NoParallelTesting(TestExclusionReason.RESOURCE_INTENSIVE), SkipTestUntil(2024, 5, 29, "UniProtKB REST API slowness issue")]
+        [TestMethod, NoParallelTesting(TestExclusionReason.RESOURCE_INTENSIVE), Timeout(TEST_TIMEOUT_MSEC)]
         public void UniquePeptides1PerfTest()
         {
             // 1)  No current background proteome
-            scenario(PeptideFilter.PeptideUniquenessConstraint.gene, null, "human_and_yeast_no_metadata.protdb", true);
+            scenario(PeptideFilter.PeptideUniquenessConstraint.gene, null, "human_and_yeast_no_metadata.protdb");
         }
 
-        [TestMethod, NoParallelTesting(TestExclusionReason.RESOURCE_INTENSIVE), SkipTestUntil(2024, 5, 29, "UniProtKB REST API slowness issue")]
+        [TestMethod, NoParallelTesting(TestExclusionReason.RESOURCE_INTENSIVE), Timeout(TEST_TIMEOUT_MSEC)]
         public void UniquePeptides2PerfTest()
         {
             // 2)  Current background proteome same as in new settings, needs digest and protein metadata search
             scenario(PeptideFilter.PeptideUniquenessConstraint.protein, "human_and_yeast_no_digest.protdb");
         }
 
-        [TestMethod, NoParallelTesting(TestExclusionReason.RESOURCE_INTENSIVE), SkipTestUntil(2024, 5, 29, "UniProtKB REST API slowness issue")]
+        [TestMethod, NoParallelTesting(TestExclusionReason.RESOURCE_INTENSIVE), Timeout(TEST_TIMEOUT_MSEC)]
         public void UniquePeptides3PerfTest()
         {
             // 3)  Current background proteome same as in new settings, needs protein metadata search
             scenario(PeptideFilter.PeptideUniquenessConstraint.gene, "human_and_yeast_no_metadata_too.protdb");
         }
 
-        [TestMethod, NoParallelTesting(TestExclusionReason.RESOURCE_INTENSIVE), SkipTestUntil(2024, 5, 29, "UniProtKB REST API slowness issue")]
+        [TestMethod, NoParallelTesting(TestExclusionReason.RESOURCE_INTENSIVE), Timeout(TEST_TIMEOUT_MSEC)]
         public void UniquePeptides4PerfTest()
         {
             // 4)  Current background proteome not same as in new settings
             scenario(PeptideFilter.PeptideUniquenessConstraint.species, "human_and_yeast.protdb", "human_and_yeast_no_metadata.protdb");
         }
 
-        [TestMethod, NoParallelTesting(TestExclusionReason.RESOURCE_INTENSIVE)]
+        [TestMethod, NoParallelTesting(TestExclusionReason.RESOURCE_INTENSIVE), Timeout(TEST_TIMEOUT_MSEC)]

```

Comment:
It would be nice if a timeout in one of these tests would cause the others get skipped (timeouts in Prosit/Koina tests would be another place that would be useful): would that be excessively complicated to implement? Perhaps some optional static flag variable in the Timeout attribute (if that's even possible)?

Reply:
Interesting idea, but probably a different attribute - I'm trying to match the standard timeout behavior in Test Explorer.